lib/firebase/user.dart

```diff
@@ -148,13 +168,16 @@ sealed class ThcUser {
       };
 
   /// Saves the current user data to Firebase.
-  Future<void> upload({bool isRegistered = true}) async {
-    final collection = isRegistered ? 'users' : 'unregistered_users';
-    await db.doc('$collection/$id').set(json);
-  }
-
-  static Future<void> removeUnregisteredUser(String id) async =>
-      await db.doc('unregistered_users/$id').delete();
```

Comment:
Instead of `removeUnregisteredUser()`, we now have a `remove()` function that can target either collection.